tests/backup_restore_test.go

```diff
@@ -138,98 +178,127 @@ func TestPITR(t *testing.T) {
+		t.Log("restore to pitr database")
+		_, err := db.Exec("SET foreign_key_checks=OFF")
+		a.NoError(err)
 
-	// concurrently update data
-	stopChan := make(chan bool)
-	go updateRow(t, username, localhost, database, port, stopChan)
-	defer func() { stopChan <- true }()
+		pitrDatabaseName := dbUtil.GetPITRDatabaseName(database)
+		_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", pitrDatabaseName))
+		a.NoError(err)
 
-	// restore to ghost database
-	_, err = db.Exec("SET foreign_key_checks=OFF")
-	a.NoError(err)
+		driverRestore := getDbDriver(t, localhost, fmt.Sprintf("%d", port), username, pitrDatabaseName)
+		defer func() {
+			err := driverRestore.Close(context.TODO())
+			a.NoError(err)
+		}()
+		err = driverRestore.Restore(context.TODO(), bufio.NewScanner(buf))
+		a.NoError(err)
 
-	_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s_ghost", database))
-	a.NoError(err)
-	_, err = db.Exec(fmt.Sprintf("USE %s_ghost", database))
-	a.NoError(err)
+		t.Log("apply binlog from t0 to t1")
+		// TODO(dragonly): implement RestoreIncremental in mysql driver
+		err = driverRestore.RestoreIncremental(context.TODO(), t0, t1)
+		a.Error(err)
 
-	driverRestore := getDbDriver(a, localhost, fmt.Sprintf("%d", port), username, fmt.Sprintf("%s_ghost", database))
-	defer func() {
-		err := driverRestore.Close(context.TODO())
+		_, err = db.Exec("SET foreign_key_checks=ON")
 		a.NoError(err)
-	}()
-	err = driverRestore.Restore(context.TODO(), bufio.NewScanner(buf))
-	a.NoError(err)
 
-	// TODO(dragonly): validate ghost table data and schema
+		t.Log("cutover stage")
+		stopChan <- true
+		time.Sleep(time.Second)
 
-	// TODO(dragonly): apply binlog from full backup to
-	// need to use binlog package from gh-ost or go-mysql
+		pitrOldDatabase := dbUtil.GetPITRDatabaseOldName(database)
+		_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s", pitrOldDatabase))
+		a.NoError(err)
 
-	_, err = db.Exec("SET foreign_key_checks=ON")
-	a.NoError(err)
+		// TODO(dragonly): extract this into a db util function, which renames all the tables in the pitr database
+		queryRename := fmt.Sprintf("RENAME TABLE"+
+			" `%s`.`t0` TO `%s`.`t0`, `%s`.`t1` TO `%s`.`t1`,"+
+			" `%s`.`t0` TO `%s`.`t0`, `%s`.`t1` TO `%s`.`t1`",
+			database, pitrOldDatabase, database, pitrOldDatabase,
+			pitrDatabaseName, database, pitrDatabaseName, database)
+		t.Log(queryRename)
+		_, err = db.Exec(queryRename)
+		a.NoError(err)
 
-	// the user can inspect the ghost table for now, and decide whether to execute the cut over
-	// cut over stage, swap tables
-	_, err = db.Exec(fmt.Sprintf("CREATE DATABASE %s_old", database))
-	a.NoError(err)
-	queryRename := fmt.Sprintf("RENAME TABLE"+
-		" `%s`.`t0` TO `%s_old`.`t0`, `%s`.`t1` TO `%s_old`.`t1`,"+
-		" `%s_ghost`.`t0` TO `%s`.`t0`, `%s_ghost`.`t1` TO `%s`.`t1`",
-		database, database, database, database, database, database, database, database)
-	t.Log(queryRename)
-	_, err = db.Exec(queryRename)
-	a.NoError(err)
+		t.Log("validate table t0")
+		{
```

Comment:
Don't add "{}" which isn't a good Go or even C-style. I know it's widely used in Kernel and many places but it should really be a separate function.